File: tv_apple/NasCabTV/Models/LoginResponse.swift
import Foundation

struct LoginResponse {
    let success: Bool
    let message: String?
    let accessToken: String?
    let refreshToken: String?
    let user: [String: Any]?
    let twoFactorRequired: Bool
    let tempToken: String?
    let platform: String?
    let hostname: String?
    let serverId: String?
    let httpPort: String?
    let httpsPort: String?
    let lanIpv4: String?
    let p2pRemoteAccessEnabled: Bool
    let pairCode: String?
    let expiresIn: Int?
    let code: Int?
    let shellSupported: Bool
    /// 服务端版本号（登录成功时返回，如 `6.0.0`）
    let serverVersion: String?

    init(
        success: Bool,
        message: String? = nil,
        accessToken: String? = nil,
        refreshToken: String? = nil,
        user: [String: Any]? = nil,
        twoFactorRequired: Bool = false,
        tempToken: String? = nil,
        platform: String? = nil,
        hostname: String? = nil,
        serverId: String? = nil,
        httpPort: String? = nil,
        httpsPort: String? = nil,
        lanIpv4: String? = nil,
        p2pRemoteAccessEnabled: Bool = false,
        pairCode: String? = nil,
        expiresIn: Int? = nil,
        code: Int? = nil,
        shellSupported: Bool = false,
        serverVersion: String? = nil
    ) {
        self.success = success
        self.message = message
        self.accessToken = accessToken
        self.refreshToken = refreshToken
        self.user = user
        self.twoFactorRequired = twoFactorRequired
        self.tempToken = tempToken
        self.platform = platform
        self.hostname = hostname
        self.serverId = serverId
        self.httpPort = httpPort
        self.httpsPort = httpsPort
        self.lanIpv4 = lanIpv4
        self.p2pRemoteAccessEnabled = p2pRemoteAccessEnabled
        self.pairCode = pairCode
        self.expiresIn = expiresIn
        self.code = code
        self.shellSupported = shellSupported
        self.serverVersion = serverVersion
    }

    static func from(json data: [String: Any], httpCode: Int) -> LoginResponse {
        LoginResponse(
            success: true,
            accessToken: data["accessToken"] as? String,
            refreshToken: data["refreshToken"] as? String,
            user: data["user"] as? [String: Any],
            twoFactorRequired: data["twoFactorRequired"] as? Bool ?? false,
            tempToken: data["tempToken"] as? String,
            platform: data["platform"] as? String,
            hostname: data["hostname"] as? String,
            serverId: data["serverId"] as? String,
            httpPort: stringValue(data["httpPort"]),
            httpsPort: stringValue(data["httpsPort"]),
            lanIpv4: stringValue(data["lanIpv4"]),
            p2pRemoteAccessEnabled: data["p2pRemoteAccessEnabled"] as? Bool ?? false,
            pairCode: stringValue(data["pairCode"]),
            expiresIn: data["expiresIn"] as? Int,
            code: httpCode,
            shellSupported: {
                if let b = data["shellSupported"] as? Bool { return b }
                if let n = data["shellSupported"] as? Int { return n == 1 }
                return false
            }(),
            serverVersion: data["serverVersion"] as? String
        )
    }

    private static func stringValue(_ v: Any?) -> String? {
        guard let v else { return nil }
        if let s = v as? String { return s }
        return "\(v)"
    }
}

struct ServerStatusResponse {
    let success: Bool
    let message: String?
    let isNasCabServer: Bool
    let serverData: [String: Any]?
}

File: flutter_client/lib/modules/auth/service/response/login_response.dart
/// 登录响应数据模型
class LoginResponse {
  final bool success;
  final String? message;
  final String? accessToken;
  final String? refreshToken;
  final Map<String, dynamic>? user;
  final bool? twoFactorRequired;
  final String? tempToken;
  final String? platform;
  final String? hostname;
  /// 服务端配置的自定义主机名（可为 null）
  final String? customHostname;
  final String? serverId;
  final String? httpPort;
  final String? httpsPort;
  final String? lanIpv4;
  final bool? p2pRemoteAccessEnabled;
  final String? pairCode;
  final int? expiresIn;
  final int? code;
  final Map<String, dynamic>? apps;
  final Map<String, dynamic>? wallpaper;
  final bool? shellSupported;

  /// 服务器当前版本（登录接口返回）
  final String? serverVersion;
  LoginResponse({
    this.code,
    required this.success,
    this.message,
    this.accessToken,
    this.refreshToken,
    this.user,
    this.twoFactorRequired,
    this.tempToken,
    this.platform,
    this.hostname,
    this.customHostname,
    this.serverId,
    this.httpPort,
    this.httpsPort,
    this.lanIpv4,
    this.p2pRemoteAccessEnabled,
    this.pairCode,
    this.expiresIn,
    this.apps,
    this.wallpaper,
    this.shellSupported,
    this.serverVersion,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json, int code) {
    // 检查是否是完整的响应格式（包含success字段）
    // 这可能是仅data字段的内容

    return LoginResponse(
      code: code,
      success: true, // 当从dataParser调用时，假设success为true
      message: null,
      accessToken: json['accessToken'],
      refreshToken: json['refreshToken'],
      user: json['user'],
      twoFactorRequired: json['twoFactorRequired'] == true,
      tempToken: json['tempToken'],
      platform: json['platform'],
      hostname: json['hostname'],
      customHostname: () {
        final raw = json['customHostname']?.toString().trim();
        if (raw == null || raw.isEmpty) return null;
        return raw;
      }(),
      serverId: json['serverId'],
      httpPort: json['httpPort'],
      httpsPort: json['httpsPort'],
      lanIpv4: json['lanIpv4']?.toString(),
      p2pRemoteAccessEnabled: json['p2pRemoteAccessEnabled'] == true,
      pairCode: json['pairCode']?.toString(),
      expiresIn: json['expiresIn'],
      apps: json['apps'],
      wallpaper: json['wallpaper'],
      shellSupported:
          json['shellSupported'] == true ||
          json['shellSupported'] == 1 ||
          json['shellSupported'] == '1',
      serverVersion: json['serverVersion']?.toString(),
    );
  }
  @override
  String toString() {
    return 'LoginResponse{success: $success, message: $message, accessToken: $accessToken, refreshToken: $refreshToken, user: $user, twoFactorRequired: $twoFactorRequired, tempToken: $tempToken, platform: $platform, hostname: $hostname, serverId: $serverId, httpPort: $httpPort, httpsPort: $httpsPort, lanIpv4: $lanIpv4, p2pRemoteAccessEnabled: $p2pRemoteAccessEnabled, pairCode: $pairCode, expiresIn: $expiresIn, apps: $apps, wallpaper: $wallpaper, shellSupported: $shellSupported}';
  }
}
